modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java

```diff
@@ -293,52 +294,79 @@
 import static org.apache.ignite.lifecycle.LifecycleEventType.BEFORE_NODE_START;
 
 /**
- * Ignite kernal.
- * <p/>
- * See <a href="http://en.wikipedia.org/wiki/Kernal">http://en.wikipedia.org/wiki/Kernal</a> for information on the
- * misspelling.
+ * This class represents an implementation of the extended main Ignite API (called {@link IgniteEx}). It also controls
+ * the Ignite life cycle and allows listening for Ignite events.
+ * <p>
+ * Please, refer to the wiki <a href="http://en.wikipedia.org/wiki/Kernal">http://en.wikipedia.org/wiki/Kernal</a>
+ * for the information on the misspelling.
+ * <p>
+ * <h3>Start point</h3>
+ * The main entry point for all the Ignite instances creation is the <tt>org.apache.ignite.internal.IgniteKernal#start()</tt>
```

Comment:
I suppose that shutdown procedure via `stop` might be mentioned here as well.

Reply:
I've added `Endpoint` block to the end of class javadoc. Please, check.